### PR TITLE
Quiz

### PR DESCRIPTION
**Pull request by tanveertkd**
Added quizzes by category. All the quizzes are displayed in the category page. It displays dummy data for now but has full functionality.
The login/sign up pages are just the screens for now and don't have any functionality to them yet

### CODE REVIEW

File: src/components/Navbar/Navbar.tsx [hunk 1]
```diff
@@ -1,31 +1,61 @@
...
+                                setTheme('dark-mode');
+                                localStorage.setItem('currentTheme', 'dark');
+                            }}
+                        />
+                    ) : (
+                        <i
+                            className="fas fa-sun nav-main-item nav-btn-login"
+                            onClick={() => {
+                                setTheme('');
+                                localStorage.setItem('currentTheme', '');
```
Comment (msdeshmukh009) [inline src/components/Navbar/Navbar.tsx:36]: You can extract this function out of your jsx.

File: src/components/QuizCard/QuizCard.tsx [hunk 2]
```diff
@@ -1,20 +1,38 @@
...
+    quiz: {
+        id: number;
+        quizTitle: string;
+        quizBrief: string;
+    };
+};
+
+const QuizCard = ({ id, quiz }: quizId) => {
+    const navigate = useNavigate();
+    const theme = localStorage.getItem('currentTheme');
```
Comment (msdeshmukh009) [inline src/components/QuizCard/QuizCard.tsx:17]: You can access the theme via theme context as well.

File: src/contexts/ThemeContext.tsx [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React, { createContext, useContext, useState } from "react";
+
+type themeType = {
+    theme: string|null;
+    setTheme: React.Dispatch<React.SetStateAction<string | null>>;
+}
+
+type childrenType = {
+    children: React.ReactNode;
+}
```
Comment (msdeshmukh009) [inline src/contexts/ThemeContext.tsx:10]: You can extract these types as well.

**Review by msdeshmukh009 — APPROVED** [review]
I think the theme is getting applied after the refresh. Other than that everything looks fine to me.

**Comment by poornima-alugubelly** [comment]
When the user selects an option , the style of the option should change indicating that the user has selected that option other than this rest is good

**Review by poornima-alugubelly — APPROVED** [review]

**Pull request merged**